src/tools/compiletest/src/runtest.rs

```diff
@@ -3660,27 +3660,35 @@ impl<'test> TestCx<'test> {
             rustc.arg("--sysroot").arg(&stage0_sysroot);
         }
 
+        // Now run rustc to build the recipe.
         let res = self.run_command_to_procres(&mut rustc);
         if !res.status.success() {
             self.fatal_proc_rec("run-make test failed: could not build `rmake.rs` recipe", &res);
         }
 
-        // FIXME(jieyouxu): explain what the hecc we are doing here.
-        let mut stage_std_path = PathBuf::new();
-        stage_std_path.push(&build_root);
-        stage_std_path.push(&stage);
-        stage_std_path.push("lib");
+        // To actually run the recipe, we have to provide the recipe with a bunch of information
+        // provided through env vars.
 
-        // FIXME(jieyouxu): explain what the hecc we are doing here.
+        // Compute stage-specific standard library paths.
+        let stage_std_path = {
```

Comment:
All these path builders in nested blocks are quite verbose. Can't we just do `let stage_std_path = build_root.join(&stage).join("lib");`? Seems easier to read to me.

Reply:
Switched to joins.